ratis-server/src/main/java/org/apache/ratis/server/impl/LeaderStateImpl.java

```diff
@@ -1113,6 +1116,12 @@ CompletableFuture<Long> getReadIndex(Long readAfterWriteConsistentIndex) {
           new LeaderNotReadyException(server.getMemberId())));
     }
 
+    // if lease is enabled, check lease first
+    if (hasLease()) {
```

Comment:
It seems that even though lease read is not enabled, we are still calling `hasLease` frequently and attempting to extend lease many times. Maybe we could use some optimization here, since it's a critical step in the query.

Reply:
You are right! Fixed, PTAL.